ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -392,18 +400,50 @@ def get_update_vars(self):
         """
         return list(self.update_dict.keys())
 
-    def save_model(self, steps):
+    def checkpoint(self, model_reward: Optional[float] = None) -> None:
         """
-        Saves the model
-        :param steps: The number of steps the model was trained for
-        :return:
+        Checkpoints the model
         """
+        current_step = self.get_current_step()
         with self.graph.as_default():
-            last_checkpoint = os.path.join(self.model_path, f"model-{steps}.ckpt")
-            self.saver.save(self.sess, last_checkpoint)
+            last_checkpoint = os.path.join(
+                self.model_path, f"model-{current_step}.ckpt"
+            )
+            if self.saver:
+                self.saver.save(self.sess, last_checkpoint)
             tf.train.write_graph(
                 self.graph, self.model_path, "raw_graph_def.pb", as_text=False
             )
+        brain_name = self.behavior_id.brain_name
+        checkpoint_path = f"{brain_name}-{current_step}"
+        settings = SerializationSettings(self.model_path, brain_name, checkpoint_path)
+        export_policy_model(settings, self.graph, self.sess, is_checkpoint=True)
+        # Store steps and file_path
+        new_checkpoint = Checkpoint(
+            int(current_step),
+            os.path.join(self.model_path, f"{settings.checkpoint_path}.nn"),
+            model_reward,
+            time.strftime("%Y-%m-%d %H:%M:%S"),
```

Comment:
Maybe change this to a Unix timestamp?  This avoids timezone issues and is what we already use server-side.
```suggestion
            int(time.time()),
```

Reply:
Good catch, didn't realize the implications of that, will change it.